adapter/src/hermes/adapter/stdio/metadata_manager.cc

```diff
@@ -0,0 +1,56 @@
+#include "metadata_manager.h"
+
+/**
+ * Namespace declarations for cleaner code.
+ */
+using hermes::adapter::stdio::AdapterStat;
+using hermes::adapter::stdio::FileID;
+using hermes::adapter::stdio::MetadataManager;
+
+bool MetadataManager::Create(FILE *fh, const AdapterStat &stat) {
+  LOG(INFO) << "Create metadata for file handler." << std::endl;
+  auto ret = metadata.emplace(Convert(fh), stat);
+  return ret.second;
+}
+
+bool MetadataManager::Update(FILE *fh, const AdapterStat &stat) {
+  LOG(INFO) << "Update metadata for file handler." << std::endl;
+  auto fileId = Convert(fh);
+  auto iter = metadata.find(fileId);
+  if (iter != metadata.end()) {
+    metadata.erase(iter);
+    auto ret = metadata.emplace(fileId, stat);
+    return ret.second;
+  } else {
+    return false;
+  }
+}
+
+std::pair<AdapterStat, bool> MetadataManager::Find(FILE *fh) {
+  auto fileId = Convert(fh);
+  typedef std::pair<AdapterStat, bool> MetadataReturn;
+  auto iter = metadata.find(fileId);
+  if (iter == metadata.end())
+    return MetadataReturn(AdapterStat(), false);
+  else
+    return MetadataReturn(iter->second, true);
+}
+
+FileID MetadataManager::Convert(FILE *fh) {
+  struct stat st;
+  int fd = fileno(fh);
+  fstat(fd, &st);
```

Comment:
Probably should check the return code of `fstat`.

Reply:
Done